llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -1474,28 +1474,22 @@ bool RegisterCoalescer::reMaterializeTrivialDef(const CoalescerPair &CP,
   //
   // The implicit-def of the super register may have been reduced to
   // subregisters depending on the uses.
-
-  bool NewMIDefinesFullReg = false;
-
-  SmallVector<MCRegister, 4> NewMIImplDefs;
+  SmallVector<std::pair<unsigned, MCRegister>, 4> NewMIImplDefs;
```

Comment:
nit: If you make the type `Register` instead of `MCRegister`, you only need a single `asMCReg`() in `TRI->regunits(Reg.asMCReg())`.